### ISSUE TITLE
numclass

### ISSUE THREAD

**Issue by dzhok** [EvilGhostY/MMRSSeg#5]
谢谢你分享这份令人印象深刻的工作！我已经读了你的论文，发现它非常有见地。
但在运行whusar数据集时有一个问题我在思考，whusar是7类别，那这段代码怎么保证整除呢
class DualChannelSplitConcat(nn.Module):
    def __init__(self, rgb_channels, dsm_channels, num_splits):
        super(DualChannelSplitConcat, self).__init__()
        assert rgb_channels % num_splits == 0, "The number of channels in the first tensor must be divisible by the number of splits."
        assert dsm_channels % num_splits == 0, "The number of channels in the first tensor must be divisible by the number of splits."

        self.num_splits = num_splits
        self.split_size1 = rgb_channels // num_splits
        self.split_size2 = dsm_channels // num_splits

    def forward(self, x1, x2):
        # 
        x1_split = torch.chunk(x1, self.num_splits, dim=1)  # (num_splits, B, split_size1, H, W)
        x2_split = torch.chunk(x2, self.num_splits, dim=1)  # (num_splits, B, split_size2, H, W)

        # 
        output_splits = []
        for i in range(self.num_splits):
            output_splits.append(x1_split[i])  #
            output_splits.append(x2_split[i])  # 

        # 
        output = torch.cat(output_splits, dim=1)

        return output

**Comment by EvilGhostY** [comment]
不好意思，最近在忙于毕业事项，才看到。
因为在解码器的时候，模型会将所有基于backbone得到的多通道特征，统一进行进行通道调整。通道数是基于类别数决定的。
如果类别数为N,那么解码器通道数就是N×32（这是可以自己设置的），所以就天然保证了通道数可以整除。
当然，这也带来天然的弱势，如果类别数过多，那么网络参数量会递增，如果不想基于类别数来设置编码器特征的通道数，则需要进行代码修改，并将CSC变成传统的concat拼接，精度影响不会很大。

**Comment by dzhok** [comment]
非常感谢作者细致的解答，解开了我的疑惑
